### ISSUE TITLE
Duplicate entry ... for key 'block_id'

### ISSUE THREAD

**Issue by Bobcat00** [AddstarMC/Prism-Bukkit#28]
Running CraftBukkit 1.12.2 with Prism downloaded from the repository as of May 30, 2018.  I'm getting repeated errors like **Database connection error: Duplicate entry '261-0' for key 'block_id'**.  Database is only a few hours old and checks out OK.  I don't know enough about MySQL to know if this is a problem in the plugin or database or something else.  (block_id is going to contain duplicates of course, so I don't even understand what the error means.)

Here's the full error:  https://pastebin.com/1NtMJkM4

In this case it's trying to write to the database, but the error also occurs on lookup commands.  Any help would be appreciated.

**Comment by MiniWorks18** [comment]
Sounds like a database issue. Check the `prism_data` properties with 
> SHOW COLUMNS FROM prism_data; 

Your table likely has the wrong Key implementation. Make sure only field `id`'s key is set to `PRI`, and that `block_id` has no Key. 
The commands to edit the keys will be something like these 

> ALTER TABLE prism_data DROP PRIMARY KEY(block_id);
ALTER TABLE prism_data ADD PRIMARY KEY(id);

**Comment by MiniWorks18** [comment]
I've been advised my fix may not be the right one. 
Could you provide the exact build versions for Spigot and Prism?

**Comment by Bobcat00** [comment]
I've looked at the Prism commits some more, and it appears that the version as of May 30, 2018 already had some changes in preparation for 1.13.  So that's the likely cause of the problem.

I compiled the version from Feb 23, 2018, commit 5b54f1a and it doesn't have the database errors.  However, it doesn't properly rollback item frames, which was the thing that was supposed to be fixed/improved by that commit.

So I guess I'm asking for some guidance of which is the proper version to use for CraftBukkit 1.12.2.  For that matter, is there a CI/Jenkins server with these builds going back for enough for use with 1.12.2?
